code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/filter/dynamicSchema/DynamicSchemaRoleFilter.java

```diff
@@ -0,0 +1,50 @@
+package io.cattle.platform.iaas.api.filter.dynamicSchema;
+
+import io.cattle.platform.core.dao.DynamicSchemaDao;
+import io.cattle.platform.core.model.DynamicSchema;
+import io.github.ibuildthecloud.gdapi.exception.ValidationErrorException;
+import io.github.ibuildthecloud.gdapi.json.JsonMapper;
+import io.github.ibuildthecloud.gdapi.model.impl.SchemaImpl;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.resource.AbstractResourceManagerFilter;
+import io.github.ibuildthecloud.gdapi.request.resource.ResourceManager;
+import io.github.ibuildthecloud.gdapi.validation.ValidationErrorCodes;
+
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+
+public class DynamicSchemaRoleFilter extends AbstractResourceManagerFilter {
+
+    private static final String DEFINITION_FIELD = "definition";
+
+    @Inject
+    DynamicSchemaDao dynamicSchemaDao;
+
+    @Inject
+    JsonMapper jsonMapper;
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Object create(String type, ApiRequest request, ResourceManager next) {
+        Map<String, Object> requestObject = (Map<String, Object>) request.getRequestObject();
+        RuntimeException exception = dynamicSchemaDao.isUnique(String.valueOf(requestObject.get("name")),
```

Comment:
Don't return a runtimeException and then check for null. Return a boolean and the raise an appropriate exception.